### PR TITLE
net: nrf_provisioning: Remove attestation token

### PR DESCRIPTION
**Pull request by juhaylinen**
Remove authentication with attestation token.
Update test cases and documentation.

### CODE REVIEW

**Review by SeppoTakalo — APPROVED** [review]

File: samples/cellular/nrf_provisioning/prj.conf [hunk 1]
```diff
@@ -74,6 +66,8 @@ CONFIG_MODEM_INFO_ADD_SIM=n
 
 # Modem key management
 CONFIG_MODEM_KEY_MGMT=y
+CONFIG_MODEM_JWT=y
+CONFIG_MODEM_ATTEST_TOKEN=y
```
Comment (melwee01) [inline samples/cellular/nrf_provisioning/prj.conf:70]: Is the sample moving from using the attestation token to JWT only? What are the implications for the user?

Also, why is the modem_attest_token config still set in this case?

I'm confused and want to make sure I understand.

Comment (juhaylinen) [reply samples/cellular/nrf_provisioning/prj.conf:70]: After this change the nrf_provisioning library will use only JWT for authentication.

CONFIG_NRF_PROVISIONING_JWT and CONFIG_NRF_PROVISIONING_ATTESTTOKEN needs to be removed from the user's application .conf file.

CONFIG_MODEM_ATTEST_TOKEN is enabled for provisioning shell command to read the attestation token from modem. The token is still used for claiming the device. It's only for testing.

**Review by jayteemo — APPROVED** [review]

**Review by jheiskan81 — APPROVED** [review]

File: samples/cellular/nrf_provisioning/prj.conf (hunk 1 above)
Comment (plskeggs) [reply samples/cellular/nrf_provisioning/prj.conf:70]: @juhaylinen if CONFIG_MODEM_ATTEST_TOKEN is only for testing, how will customers using this service in production gather the identity attestation tokens for later claiming?  I realize this is out of scope of this PR, but curious what the plan is.

**Review by plskeggs — APPROVED** [review]
@juhaylinen have you tested this with an unclaimed device and confirmed it can be onboarded to nRF Cloud?

**Review by melwee01 — APPROVED** [review]

File: samples/cellular/nrf_provisioning/prj.conf (hunk 1 above)
Comment (juhaylinen) [reply samples/cellular/nrf_provisioning/prj.conf:70]: I don't know what the plan is but it's possible to use AT command to read the identity attestation token.

**Comment by juhaylinen** [comment]
> @juhaylinen have you tested this with an unclaimed device and confirmed it can be onboarded to nRF Cloud?

No, I haven't tested full onboarding with nRF Cloud yet.

**Pull request merged**